sotodlib/io/hkdb.py

```diff
@@ -0,0 +1,425 @@
+"""
+Module for loading L3 housekeeping data.
+"""
+from dataclasses import dataclass, field
+import yaml
+import os
+import numpy as np
+
+import sqlalchemy as db
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker, relationship, backref
+from typing import Union, Tuple, List, Optional, Dict
+import so3g
+from spt3g import core as spt3g_core
+from tqdm.auto import tqdm, trange
+import time
+
+Base = declarative_base()
+Session = sessionmaker()
+
+@dataclass
+class HkConfig:
+    """
+    Configuration object for indexing and loading from an HK archive.
+    """
+    hk_root: str
+    "Root directory for the HK archive"
+
+    hk_db: str
+    "Path to the hk index database"
+
+    echo_db: bool = False
+    "Whether database operations should be echoed"
+
+    file_idx_lookback_time: Optional[float] = None
+    "Time [sec] to look back when scanning for new files to index"
+
+    show_index_pb: bool = True
+    "If true, shows progress bar when indexing"
+
+    aliases: Dict[str, str] = field(default_factory=dict)
+    """
+    Aliases for hk fields. In this dict, the key is the alias name, and the
+    value is the field descriptor, in the format of ``agent.feed.field``.
+    For example::
+
+        {
+            'fp_temp': 'cryo-ls372-lsa21yc.temperatures.Channel_02_T',
```

Comment:
Perhaps warn the user that aliases should be valid python attribute names, since you setattr with these, later.  (I.e. they should not use 'fp-temp'.)